agent/fw_drupal.c

```diff
@@ -588,6 +603,47 @@ NR_PHP_WRAPPER(nr_drupal_wrap_module_invoke) {
 }
 NR_PHP_WRAPPER_END
 
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
+     && !defined OVERWRITE_ZEND_EXECUTE_DATA
+NR_PHP_WRAPPER(nr_drupal_wrap_module_invoke_all_before) {
+  (void)wraprec;
+  zval* hook = NULL;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_DRUPAL);
+
+  hook = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+  if (!nr_php_is_zval_non_empty_string(hook)) {
+    goto leave;
+  }
+
+  zval* hook_copy = nr_php_zval_alloc();
+  if (NULL == hook_copy) {
+    goto leave;
+  }
+
+  ZVAL_DUP(hook_copy, hook);
+  if (!nr_php_is_zval_non_empty_string(hook_copy)) {
+    nrl_verbosedebug(NRL_FRAMEWORK,
+                     "%s: could not copy hook zval",
+                     __func__);
+    goto leave;
+  }
```

Comment:
I called `nr_php_arg_release` because, while the after function doesn't directly access the args, the value on the stack that it is freeing is the direct result from `nr_php_arg_get`